charts/alloy/charts/alloy-api/templates/secret.yaml

```diff
@@ -7,5 +7,5 @@ metadata:
 type: Opaque
 stringData:
 {{- range $key, $val := .Values.env }}
-  {{ $key }}: {{ $val | quote }}
+  {{ $key }}: {{ (tpl ($val | toString) $) | quote }}
```

Comment:
Good idea using `toString` here. We should apply the same to other charts that are templating secrets from values.